_search-plugins/searching-data/search-shard-routing.md

```diff
@@ -0,0 +1,212 @@
+
+The following is an example index operation with a routing value:
+
+```json
+POST /index1/_doc/1?routing=user1
+{
+  "name": "John Doe",
+  "age": 20
+}
+```
+{% include copy-curl.html %}
+
+In this example, the document with ID 1 is indexed with the routing value user1. All documents with the same routing value will be stored on the same shard.
+
+### Routing during searches
+
+When you search for documents, specifying the same routing value ensures that the search request is routed to the appropriate shard. This can significantly improve performance by reducing the number of shards that need to be queried.
+
+Here’s an example of searching with a specific routing value:
+
+```json
+GET /index1/_search?routing=user1
+{
+  "query": {
+    "match": {
+      "name": "John Doe"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In this example, the search query is routed to the shard that contains documents indexed with the routing value `user1`.
+
+Additional consideration needs to be taken when using custom routing in order to prevent hot spots and data skew:
+
+ - A _hot spot_ occurs when a disproportionate number of documents are routed to a single shard. This can lead to that shard becoming a bottleneck, because it will have to handle more read and write operations compared to other shards. Consequently, this shard may experience higher CPU, memory, and I/O usage, leading to performance degradation.
+
+ - _Data skew_ refers to an uneven distribution of data across shards. If routing values are not evenly distributed, some shards may end up storing significantly more data than others. This can result in an imbalanced storage use, where certain nodes have a much higher disk utilization compared to others.
+
+## Concurrent shard request
+
+Hitting a large number of shards simultaneously during a search can significantly impact the CPU and memory consumption. By default, OpenSearch does not reject these requests. However, there are number of methods that you can use to mitigate this risk. The following sections describe these methods.
+
+### Limit the number of shards that can be queried
+
+You can use the `max_concurrent_shard_requests` parameter in the search request to limit it to a maximum number of shards that can be queried. For example, the following request limits the number of concurrent shard requests to 12:
+
+```json
+GET /index1/_search?max_concurrent_shard_requests=12
+{
+  "query": {
+    "match_all": {}
+  }
+}
+```
+{% include copy-curl.html %}
+
+
+### Define a search shard count limit
+
+You can define the dynamic `action.search.shard_count.limit` setting in `opensearch.yml` file or using the REST API. Any search request that attempts to query the number of shards that exceeds this limit will be rejected with an error. This helps prevent a single search request from consuming too many resources, which can degrade the performance of the entire cluster. The following example updates this cluster setting using the API:
```

Comment:
```suggestion
You can define the dynamic `action.search.shard_count.limit` setting either in your `opensearch.yml` file or by using the REST API. Any search request that exceeds this limit will be rejected with an error. This helps to prevent a single search request from consuming too many resources, which can degrade the performance of the entire cluster. The following example request updates this cluster setting using the API:
```